gslab_scons/tests/test_build_matlab.py

```diff
@@ -1,50 +1,104 @@
-#! /usr/bin/env matlab
 import unittest
 import sys
 import os
 import shutil
+import mock
+import re
+# Import gslab_scons testing helper modules
+import _test_helpers as helpers
+import _side_effects as fx
 
 sys.path.append('../..')
-from gslab_scons import build_matlab
+import gslab_scons as gs
 from gslab_scons._exception_classes import BadExtensionError
 
+# Define main test patch
+path  = 'gslab_scons.builders.build_matlab'
+system_patch = mock.patch('%s.os.system'   % path)
+copy_patch   = mock.patch('%s.shutil.copy' % path)
+main_patch   = lambda f: system_patch(copy_patch(f))
 
-class testbuild_matlab(unittest.TestCase):
-
-    def setUp(self):
-        if not os.path.exists('../build/'):
-            os.mkdir('../build/')
-
-    def test_default(self):
-        env = {}
-        build_matlab('../build/test.mat', './input/matlab_test_script.m', env)
-        logfile_data = open('../build/sconscript.log', 'rU').read()
-        self.assertIn('Log created:', logfile_data)
-        self.assertTrue(os.path.isfile('../build/test.mat'))
-        if os.path.isfile('../build/sconscript.log'):
-            os.remove('../build/sconscript.log')
-     
-    def test_clarg(self):
-        env = {'CL_ARG' : 'COMMANDLINE'}
-        build_matlab('../build/testCOMMANDLINE.mat', './input/matlab_test_script.m', env)
-        self.assertTrue(os.path.isfile('../build/testCOMMANDLINE.mat'))
-        if os.path.isfile('../build/sconscript.log'):
-            os.remove('../build/sconscript.log')
-
-    def test_bad_extension(self):
-        env = {}
-        with self.assertRaises(BadExtensionError):
-            build_matlab('../build/test.mat', 
-                         ['bad', './input/matlab_test_script.m'], 
-                         env)
-   
-    def tearDown(self):
-        if os.path.exists('../build/'):
-            shutil.rmtree('../build/')
-        if os.path.exists('output.txt'):
-            os.remove('output.txt')
 
+class TestBuildMatlab(unittest.TestCase):
+
+    @helpers.platform_patch('darwin', path)
+    @main_patch
+    def test_unix(self, mock_copy, mock_system):
+        '''
+        Test that build_matlab() creates a log and properly submits
+        a matlab system command on a Unix machine.
+        '''
+        # Mock copy so that it just creates the destination file
+        mock_copy.side_effect = \
```

Comment:
Is there a way to define these side effects once? (This comment applies throughout this and other scripts.)